File: python/gate_six.py/task_one.py

def create_parking_slots():

    parking_slots =[]

    for slot_index in range(0, 20):
        parking_slots.append(0)

    return parking_slots


def parking(parking_slots, desired_parking_slot):
    
    desired_slot_status =parking_slots[desired_parking_slot-1]

    if (desired_slot_status ==0) :
        parking_slots[desired_parking_slot-1] =1
        print("You are successfully parked")
        print(f"Slot {desired_parking_slot} -->{parking_slots[desired_parking_slot-1]}")

    else:
        print(" Slot is currently occupied!! ")
        print("Action Failed!!")
 
    return parking_slots     

def unparking(parking_slots, alloted_parking_slot):

    alloted_parking_slot_status =parking_slots[alloted_parking_slot -1]

    if (alloted_parking_slot_status ==1) :
        parking_slots[alloted_parking_slot -1] =0
        print("You have successfully unparked")
        print(f"Slot {alloted_parking_slot} -->{parking_slots[alloted_parking_slot-1]}")

    else :
        print("Slot is empty!!(Are you sure this is where you parked)")
        print("Action Failed!!")

    return parking_slots

def check_slots_statuses(parking_slots):


    for slot_index in range(0, 20):
        print(f"Slot {slot_index+1} -->{parking_slots[slot_index]}     ", end=" ")
        if slot_index ==19:
            return "It works"

File: Java/GateSix.java/TaskOne.java
public class TaskOne{


    public static int[] createParkingSlots(){

        int [] parkingSlots =new int[20];
        
        for(int slotIndex =0; slotIndex <20; slotIndex++){
            
            parkingSlots[slotIndex] =0;

        }    
        return parkingSlots;

    }


    public static int[] parking(int [] parkingSlots, int desiredParkingSlot){
        
        int desiredParkingSlotStatus =parkingSlots[desiredParkingSlot-1];

        if (desiredParkingSlotStatus ==0){
            parkingSlots[desiredParkingSlot-1] =1;
            System.out.println("You are successfully parked");
            System.out.println("Slot " + desiredParkingSlot + "-->" + parkingSlots[desiredParkingSlot-1]);

        }

        else{
            System.out.println(" Slot is currently occupied!! ");
            System.out.println("Action Failed!!");
        }
        return parkingSlots;
    }




    public static int[] unparking(int [] parkingSlots, int allotedParkingSlot){

        int allotedParkingSlotStatus =parkingSlots[allotedParkingSlot-1];

        if (allotedParkingSlotStatus ==1){
            parkingSlots[allotedParkingSlot-1] =0;
            System.out.println("You have successfully unparked");
            System.out.println("Slot " + allotedParkingSlot + "-->" + parkingSlots[allotedParkingSlot-1]);

        }    
 

        else{
            System.out.println(" Slot is currently occupied!! ");
            System.out.println("Action Failed!!");
        }

        return parkingSlots;
       
    }


    public static String checkSlotsStatuses(int [] parkingSlots){

        for(int slotIndex =0; slotIndex <20; slotIndex++){

            System.out.println("Slot " + slotIndex+1 + "-->" + (parkingSlots[slotIndex]));
            if (slotIndex ==19){
                return "It works";
            }
 
       }
        return "It don't work";
    }



}
